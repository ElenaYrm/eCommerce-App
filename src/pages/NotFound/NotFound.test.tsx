import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { NotFound } from './index';
import { ReactElement } from 'react';

jest.mock('../../constant/metaData', () => {});

describe('Test NotFound page: ', () => {
  test('should render NotFound page', () => {
    const component = render(
      <MemoryRouter initialEntries={['/notfound']}>
        <NotFound />
      </MemoryRouter>,
    );
    expect(screen.getByText('Page not found (´•̥̥̥o•̥̥̥`)')).toBeInTheDocument();
    expect(component).toMatchSnapshot();
  });

  test('should redirect to Main page on link click', async () => {
    const HomePage = (): ReactElement => <div>Main page</div>;

    render(
      <MemoryRouter initialEntries={['/notfound']}>
        <Routes>
          <Route path="/*" element={<NotFound />} />
          <Route path="/" element={<HomePage />} />
        </Routes>
      </MemoryRouter>,
    );

    const mainLink = screen.getByText('Back to Main');
    expect(mainLink).toBeInTheDocument();

    await userEvent.click(mainLink);
    expect(screen.getByText('Main page')).toBeInTheDocument();
  });
});
