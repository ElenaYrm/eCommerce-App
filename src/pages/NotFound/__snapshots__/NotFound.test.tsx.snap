// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test NotFound page:  should render NotFound page 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="notFound"
      >
        <div
          class="notFound__content"
        >
          <h2>
            Page not found (´•̥̥̥o•̥̥̥\`)
          </h2>
          <p>
            That's sad... But you know you are always welcome on the main page, right?
          </p>
          <a
            class="notFound__btn"
            href="/"
          >
            Back to Main
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="notFound"
    >
      <div
        class="notFound__content"
      >
        <h2>
          Page not found (´•̥̥̥o•̥̥̥\`)
        </h2>
        <p>
          That's sad... But you know you are always welcome on the main page, right?
        </p>
        <a
          class="notFound__btn"
          href="/"
        >
          Back to Main
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
